Damage types 3

Humonitarian opened the pull request:
## About The Pull Request

Remake of the PR

## Testing

We might need a testserver again.

## Changelog
:cl:
code: damage types and EE
/:cl:

SirRichardFrancis commented:
Came to check on how Eris is doing code-wise, opened first PR in the list and… what am I seeing exactly?

5k diffs, no description, "test it for me", clearly balance-related… hahaclassic.png

Skimmed trough diffs, and it's immediately obvious that code is not compliant with guidelines and lots of numbers have been changed arbitrary.

You also didn't bother with the "Why It's Good For The Game" section, let me add it for you.

## Why It's Good For The Game

It isn't.

Mycah142 commented:
It's because you lack context. The PR previously existed and was merged, but because of a fundamental error in the subsystem it was tied to, both were reverted with intention for this PR to be reintroduced, Humon just didn't have time on Friday as Fox did the Thanos snap during Merge Time before going live.

Your vitriol is noticed, but if you're just going to have disdain for something you decided to not actually look into more than skimming, you're not in the right. You're just an asshole.

Humonitarian commented:
> Came to check on how Eris is doing code-wise, opened first PR in the list and… what am I seeing exactly?
> 
> 5k diffs, no description, "test it for me", clearly balance-related… hahaclassic.png
> 
> Skimmed trough diffs, and it's immediately obvious that code is not compliant with guidelines and lots of numbers have been changed arbitrary.
> 
> You also didn't bother with the "Why It's Good For The Game" section, let me add it for you.
> 
> ## Why It's Good For The Game
> It isn't.

About The Pull Request
Implements damage type handling, first half of the melee damage rebalance.

Includes taking into account structures and slimes.

Support for FBPs is not currently made (mostly for balance reasons), they are treated as humans.

Why It's Good For The Game
Natively helps balance recursive explosions, creates foundation for the melee damage rebalance that'll add more depth to melee combat.

Testing
Had been on test server, was merged and functioned correctly on live server.

Changelog
🆑
fix: fixed a bug with armor feedback messages
fix: fixed a bug related to mechs
code: damage type system added
balance: walls now take less damage from bullets
/🆑

Humonitarian commented:
> It's because you lack context. The PR previously existed and was merged, but because of a fundamental error in the subsystem it was tied to, both were reverted with intention for this PR to be reintroduced, Humon just didn't have time on Friday as Fox did the Thanos snap during Merge Time before going live.
> 
> Your vitriol is noticed, but if you're just going to have disdain for something you decided to not actually look into more than skimming, you're not in the right. You're just an asshole.

Error was in a separate subsystem, they just conflicted and this pr prevented a revert of the other subsystem.

Humonitarian commented:
> not to mention following code standards this project meant to have

Which changes specifically? If you see some bad code habits there, please do share so I can patch them out.

hyperioo commented:
Let's see how it goes this time
